lib/site_prism/addressable_url_matcher.rb

```diff
@@ -72,29 +72,33 @@ def extract_component_templates
           component_url = component_url.sub(substituted_value, template_value)
         end
 
-        component_templates[component] = Addressable::Template.new(component_url.to_s)
+        component_templates[component] =
+          Addressable::Template.new(component_url.to_s)
       end
     end
 
-    # Returns empty hash if the template omits the component, a set of substitutions if the
-    # provided URI component matches the template component, or nil if the match fails.
+    # Returns empty hash if the template omits the component,
+    # a set of substitutions if the
+    # provided URI component matches the template component,
+    # or nil if the match fails.
     def component_matches(component, uri)
-      extracted_mappings = {}
-      component_template = component_templates[component]
-      if component_template
+      mappings = {}
+      if (component_template = component_templates[component])
         component_url = uri.public_send(component).to_s
-        extracted_mappings = component_template.extract(component_url)
-        unless extracted_mappings
+        mappings = component_template.extract(component_url)
+        unless mappings
           # to support Addressable's expansion of queries
           # ensure it's parsing the fragment as appropriate (e.g. {?params*})
           prefix = COMPONENT_PREFIXES[component]
-          return nil unless prefix && (extracted_mappings = component_template.extract(prefix + component_url))
+          return nil unless prefix
```

Comment:
Is this line the same (And some weird ruby-ness I don't know)

Reply:
original condition `return nil if prefix is nil` or `component_template.extract(prefix + component_url) is nil`. In the first case we can immediately return nil (new line 93). in second case mapping will be nil and we return nil at the end of the method (line 97)

Reply:
Yes... we can... Top spot!